### ISSUE TITLE
Add support for sharing a value directly from a notebook input

### ISSUE THREAD

**Issue by jonsequitur** [dotnet/interactive#2595]
Currently, using an `@input:name` token in a magic command, it's possible to request user input to store a variable in the `#!value` kernel and then share it using `#!share` into e.g. the C# kernel. The code looks like this:

```console
#!value --from-value @input:"Please enter the value for parameter1" --name parameter1

#!csharp
#!share --from value parameter1
```  

This is very verbose, so it would be useful to have a more direct way to share into a given kernel (other than #!value, which has this capability already using the `--from-value` option).

Here are a few examples for the sake of discussion.

One approach could be to add options to the `#!share` magic:

```console
#!share --from-value @input:"Please enter the value for parameter1"  parameter1
```

The use of the word "share" seems a little off here, though. The final token in the example, `parameter`, is the name of the variable, which today is required and references the variable name in the source kernel. Making it non-required could make a slightly more readable approach possible:

```console
#!share --from-value @input:"Please enter the value for parameter1"  --as parameter1
```

Another option could be to introduce a new magic altogether, for example:

```console
#!declare --value @input:"Please enter the value for parameter1" --name parameter1
```

In this case, adding support for type hints might also be useful, to provide more specific UI (file choosers, quick picks) and some degree of validation:

```console
#!declare --value @input:"Please enter the value for parameter1" --name parameter1 --type file
```

**Comment by dgorti** [comment]
You can apparently do this via 
`var i = await Kernel.GetInputAsync("Give Some input", "here");`

**Comment by dgorti** [comment]
If we need to get multiple values, getting input prompt for each variable can get tedious. 
May I propose that perhaps some kind of simple HTML form with a set of inputs? The return value could be bound to a json value that can be parsed later.

**Comment by jonsequitur** [comment]
We discussed an enhancement to the current design that doesn't require an additional kernel or syntax. We could detect when there are multiple input tokens in a magic command and generate a form instead of an input-by-input prompt.

**Comment by jonsequitur** [comment]
I'm closing this issue because it's been addressed by #2652 (a more specific design that addresses this scenario) and #2654 (the PR implementing it).
